integrations/azure_ai_search/tests/test_document_store.py

```diff
@@ -0,0 +1,365 @@
+                        "number": -10,
+                        "date": "1989-11-09T17:53:00Z",
+                    },
+                    embedding=_random_embeddings(768),
+                )
+            )
+
+            documents.append(
+                Document(content=f"Doc {i} with zeros emb", meta={"name": "zeros_doc"}, embedding=TEST_EMBEDDING_1)
+            )
+            documents.append(
+                Document(content=f"Doc {i} with ones emb", meta={"name": "ones_doc"}, embedding=TEST_EMBEDDING_2)
+            )
+        return documents
+
+    # Overriding to compare the documents with the same order
+    def assert_documents_are_equal(self, received: List[Document], expected: List[Document]):
+        """
+        Assert that two lists of Documents are equal.
+
+        This is used in every test, if a Document Store implementation has a different behaviour
+        it should override this method. This can happen for example when the Document Store sets
+        a score to returned Documents. Since we can't know what the score will be, we can't compare
+        the Documents reliably.
+        """
+        sorted_recieved = sorted(received, key=lambda doc: doc.id)
+        sorted_expected = sorted(expected, key=lambda doc: doc.id)
+        assert sorted_recieved == sorted_expected
+
+    # Dataframes are not supported in serach index
+    def test_comparison_equal_with_dataframe(self, document_store, filterable_docs):
+        pass
+
+    def test_comparison_not_equal_with_dataframe(self, document_store, filterable_docs):
+        pass
+
+    def test_comparison_greater_than_with_dataframe(self, document_store, filterable_docs):
+        pass
+
+    def test_comparison_less_than_with_dataframe(self, document_store, filterable_docs):
+        pass
+
+    def test_comparison_greater_than_equal_with_dataframe(self, document_store, filterable_docs):
+        pass
+
+    def test_comparison_less_than_equal_with_dataframe(self, document_store, filterable_docs):
+        pass
+
+    # Azure search index supports UTC datetime in ISO format
+    def test_comparison_greater_than_with_iso_date(self, document_store, filterable_docs):
+        """Test filter_documents() with > comparator and datetime"""
+        document_store.write_documents(filterable_docs)
+        result = document_store.filter_documents(
+            {"field": "meta.date", "operator": ">", "value": "1972-12-11T19:54:58Z"}
+        )
+        self.assert_documents_are_equal(
+            result,
+            [
+                d
+                for d in filterable_docs
+                if d.meta.get("date") is not None
+                and datetime.strptime(d.meta["date"], "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc)
+                > datetime.strptime("1972-12-11T19:54:58Z", "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc)
+            ],
+        )
+
+    def test_comparison_greater_than_equal_with_iso_date(self, document_store, filterable_docs):
```

Comment:
Why do we need to override the original test?

Reply:
Mentioned the reason [above](https://github.com/deepset-ai/haystack-core-integrations/blob/23d1e2298ab620b421e063ec359f71ea6723b365/integrations/azure_ai_search/tests/test_document_store.py#L246). Although I think I could explain it better by specifically mentioning **ISO 8601 format**.